### PR TITLE
implement serde for interpreter

### PR DESCRIPTION
**Pull request by FredCoen**
I tried to tackle https://github.com/bluealloy/revm/issues/1890

I am not sure this is correct. To my understanding the instruction_pointer is just a pointer and so i have replaced it with the program counter for serialization which i know is not the same but i also wonder what the value of that pointer is in a different context. To my understanding doing an absolute jump back to that offset should give us the correct instruction_pointer again when deserializing.

I probably need some guidance here, if this is too far from expecatiations please let me know i can also just let it go :)

In case this is correct i probably have to add a test where the instruction_pointer changes i.e a running execution. What is the best way to mock the host, do i have to implement a transaction, block etc. from scratch? @rakita thanks!

### CODE REVIEW

File: crates/interpreter/src/interpreter/ext_bytecode/serde.rs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+use crate::interpreter::Jumps;
+
```
Comment (rakita) [inline crates/interpreter/src/interpreter/ext_bytecode/serde.rs:2]:
```suggestion
```

fmt after this

File: crates/interpreter/src/interpreter/ext_bytecode/serde.rs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    where
+        D: Deserializer<'de>,
+    {
+        let ExtBytecodeSerde {
+            base,
+            program_counter,
+        } = ExtBytecodeSerde::deserialize(deserializer)?;
+
+        let mut bytecode = Self::new(base);
+        bytecode.absolute_jump(program_counter);
```
Comment (rakita) [inline crates/interpreter/src/interpreter/ext_bytecode/serde.rs:36]: Would put a small sanity check above this line, `if program_counter < bytecode.bytecode_len() { panic!("serde pc: {program_counter} is less than bytecode len");`

**Review by rakita — APPROVED** [review]
Few nits, othrwise lgtm

File: crates/interpreter/src/interpreter/ext_bytecode/serde.rs (hunk 2 above)
Comment (FredCoen) [reply crates/interpreter/src/interpreter/ext_bytecode/serde.rs:39]: added!

i believe you meant the message to be "\ is greater than or equal to bytecode len"

Comment (rakita) [reply crates/interpreter/src/interpreter/ext_bytecode/serde.rs:39]:
```
i believe you meant the message to be " is greater than or equal to bytecode len"
```
Yeah, i edited my msg after i have reread it :D

**Review by rakita — APPROVED** [review]
lgtm

**Pull request merged**